src/middleware/bodyParser/formData.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Port of body-parser (https://github.com/expressjs/body-parser) for Deno.
+ *
+ * Licensed as follows:
+ *
+ * (The MIT License)
+ *
+ * Copyright (c) 2014 Jonathan Ong <me@jongleberry.com>
+ * Copyright (c) 2014-2015 Douglas Christopher Wilson <doug@somethingdoug.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * 'Software'), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+import type { NextFunction, OpineRequest, OpineResponse } from "../../types.ts";
+import { hasBody, MultipartReader } from "../../../deps.ts";
+import { typeChecker } from "./typeChecker.ts";
+
+/**
+ * Create a middleware to parse formData.
+ *
+ * @param {object} [options]
+ * @return {function}
+ * @public
+ */
+export function formData(options: any = {}) {
+  const type = options.type || "multipart/form-data";
+  const verify = options.verify || false;
+
+  if (verify !== false && typeof verify !== "function") {
+    throw new TypeError("option verify must be function");
+  }
+
+  // create the appropriate type checking function
+  const shouldParse = typeof type !== "function" ? typeChecker(type) : type;
+
+  return function formParser(
+    req: OpineRequest,
+    _res: OpineResponse,
+    next: NextFunction,
+  ) {
+    if (req._parsedBody) {
+      next();
+
+      return;
+    }
+
+    // skip requests without bodies
+    if (
+      !hasBody(req.headers) ||
+      parseInt(req.headers.get("content-length") || "") === 0
+    ) {
+      req.parsedBody = new FormData();
+      req.body = req.parsedBody;
+
+      next();
+
+      return;
+    }
+
+    // determine if request should be parsed
+    if (!shouldParse(req)) {
+      next();
+
+      return;
+    }
+
```

Comment:
Suspect would be good to have a check for the Content-Disposition header, see https://datatracker.ietf.org/doc/html/rfc7578#section-4.2.

Reply:
https://deno.land/std@0.126.0/mime/multipart.ts#L203 multipartreader indeed handles Content-Disposition